core/snippets/README.md

```diff
@@ -0,0 +1,55 @@
+# Snippets
+
+Custom formats to represent snippets.
+
+## Features
+
+- Custom file ending `.snippet`.
+- VSCode syntax highlights.
+- VSCode document formatter.
+- Support for insertion and wrapper snippets.
+- Support for phrase formatters.
+
+## Format
+
+- A `.snippet` file can contain multiple snippet documents separated by `---`.
+- Each snippet document has a context and body separated by `-`.
+- Optionally a file can have a single context at the top with no body. This is not a snippet in itself, but default values to be inherited by the other snippet documents in the same file.
+- Some context keys supports multiple values. These values are separated by `|`.
+
+### Context fields
+
+| Key            | Required | Multiple values | Example                        |
+| -------------- | -------- | --------------- | ------------------------------ |
+| name           | Yes      | No              | `name: ifStatement`            |
+| language       | No       | Yes             | `language: javascript \| java` |
+| phrase         | No       | Yes             | `phrase: if \| if state`       |
+| insertionScope | No       | Yes             | `insertionScope: statement`    |
+
+### Variables
+
+Tab stops(`$0`) and variables(`$try`) can be used to wrap with using the following fields.
+
+| Key                | Required | Multiple values | Example                             |
+| ------------------ | -------- | --------------- | ----------------------------------- |
+| insertionFormatter | No       | Yes             | `$0.insertionFormatter: SNAKE_CASE` |
```

Comment:
It just joins them together with a comma.

Reply:
I wonder if it's worth having a note column or calling this out in a bullet point above. The semantics of that `|` are pretty dramatically different from the semantics of the `|` in all the other fields where you support it

Reply:
Or we as remove the pipe operator for formatters? They can still add it with a comma themselves if they want.

Reply:
Interesting idea. It's funny because I think the pipe is a clearer syntax than comma for those semantics with formatters, but given we already use the comma syntax for a formatter pipeline there and the overloaded semantics of pipe here, I would lean very very slightly towards comma. But I don't feel strongly. Either way it's worth a comment

Reply:
@knausj85 Any ideas?

Reply:
I think it's fine as-is.

Reply:
I'll still push gently for a comment above or a notes column to point out the semantic difference between the use of pipe for formatter vs the others

Otherwise, I think we're good to go! Shall I pull it and give it a quick test locally?

Reply:
Please do

Reply:
Readme updated.